service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -470,11 +473,90 @@ public ReportWithLibrary createReportWithLibrary(
         req.getParentSupport());
   }
 
+  @Override
+  public ReportGeneric getMostRecentReport(JWTData userData, int schoolId) {
+    SchoolsRecord school = this.queryForSchool(schoolId);
+    if (school == null) {
+      throw new SchoolDoesNotExistException(schoolId);
+    }
+
+    LibraryStatus libraryStatus = school.getLibraryStatus();
+
+    if (libraryStatus == LibraryStatus.EXISTS) {
+      return db.select(
```

Comment:
What happens in the case that the database for some reason does not find the report? I think the data that the ```db.select``` returns should be saved in a variable and then check to see if ```variable == null```

Reply:
@jtavera235 makes a good point--for all three, we should check if it's `null`, and throw a `NoReportFoundException` *and log an error to Slack* (since we should always have a report in the table if `library_status` is set on the `schools` record)